Grocery tasks

EdanSneh opened the pull request:
RAFCON w/f for picking and placing

EdanSneh committed:
Initialized rafcon package

Added env hooks and root_state

EdanSneh committed:
Creates grocery state machine init

using rafcon we added in the different states necessary to grab and
object and leave with it. The code itself was not added only the states

EdanSneh committed:
fixes RAFCON_LIBRARY_PATH

this env variable was copying multiple times every time you source it
and was just really ugly

EdanSneh committed:
Creates a state machine from grasping to shelf (no placing)

While the state machine has all the correct methods in place every
method has a bunch of underlying bugs mostly due to problems in the
simulator. Therefore the simulator will run but not run well :(

nickswalker committed:
Update RAFCON PATH insertion conditions

Previous condition wasn't working in my environment, but this recommendation did
https://unix.stackexchange.com/questions/217622/add-path-to-path-if-not-already-in-path
Use double bracket in conditional to avoid emitting error
Quote strings to handle unusual characters in variables

EdanSneh committed:
moves tuck to right place

EdanSneh committed:
Merge branch 'grocery_tasks' of https://github.com/homeskies/uw_robocup into grocery_tasks

nickswalker commented on README.md:
```diff
@@ -1,3 +1,6 @@
 
 RoboCup task specific code. Each task lives in a separate package. Shared code lives in `robot_states`.
+
+RAFCON
```
Should be a header `##`

nickswalker commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+{
```
This shouldn't be tracked

nickswalker commented on common_states/env-hooks/common_states.sh:
```diff
@@ -1 +1,3 @@
-export RAFCON_LIBRARY_PATH=$(rospack find common_states)/states/common_states:$RAFCON_LIBRARY_PATH
\ No newline at end of file
+if [[ "$RAFCON_LIBRARY_PATH" != *":$(rospack find common_states)/states/common_states:"* ]]; then
```
This is working in zsh but producing a lot of output during builds now. I'll look to see if there's a cross shell way to fix that

nickswalker commented on common_states/states/common_states/initialize_robot/initialize_robot_UYWXTI/script.py:
```diff
@@ -0,0 +1,49 @@
+        
+            fetch = (Base(), arm, Gripper(), GraspingClient(), Head(), torso)
+            gvm.set_variable("robot", fetch, per_reference=True)
+            gvm.set_variable("rospy", rospy, per_reference = True)
```
This looks suprising, because other parts of the state machine can import most other modules themselves. I assume rospy has some strange side effect? What does keeping the module around do? Do leave a comment describing why this needs to be here

nickswalker commented on common_states/states/common_states/locate_object/locate_object_ESCGUK/script.py:
```diff
@@ -0,0 +1,23 @@
+
+def execute(self, inputs, outputs, gvm):
+    fetch = gvm.get_variable("robot")
+    rospy = gvm.get_variable("rospy")
```
What happens if rospy is imported normally?

nickswalker commented on common_states/states/common_states/locate_object/locate_object_ESCGUK/script.py:
```diff
@@ -0,0 +1,23 @@
+    self.logger.info("looked at object")
+    self.logger.info("raising torso")
+    torso.set_height(.4)
+    while not rospy.is_shutdown():
```
Some use cases probably require control over how many times reperception happens, possibly with other actions happening before retrying (e.g. moving the head), so it doesn't make sense to retry forever. I would make `retries` an int parameter of the state and have it default to 3, that way it can be used with no retries in other situations.

nickswalker commented on common_states/states/common_states/move_arm/move_arm_ZFOIXZ/script.py:
```diff
@@ -0,0 +1,16 @@
+def execute(self, inputs, outputs, gvm):
+    fetch = gvm.get_variable("robot")
+    rospy = gvm.get_variable("rospy")
```
What happens if this rospy is removed, since it isn't used in the file?

nickswalker commented on task_storing_groceries/scripts/run_task:
```diff
@@ -1 +1,3 @@
-#!/usr/bin/env python
\ No newline at end of file
+#!/usr/bin/env python
```
This is a bash script

nickswalker commented on task_storing_groceries/package.xml:
```diff
@@ -9,7 +9,11 @@
\ No newline at end of file
+  <exec_depend>common_states</exec_depend>
+  <exec_depend>rafcon-pip</exec_depend>
+  <exec_depend>rospy_message_converter</exec_depend>
```
Cool

nickswalker commented on inspection/CMakeLists.txt:
```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 2.8.3)
```
It's annoying that this is picking up inspection stuff as well. I think merging master into this branch to get rid of that.

nickswalker commented on task_storing_groceries/scripts/demo.py:
```diff
@@ -0,0 +1,353 @@
+#! /usr/bin/env python
```
Is this script working or for development use? From first glance it looks like it shouldn't be tracked due to the commented out chunks.

nickswalker commented on task_storing_groceries/input_values/pose.txt:
```diff
@@ -0,0 +1 @@
+{'header': {'frame_id': 'base_link'},'pose': {'position': {'x': 0.5,'y': 0,'z': 0.4},'orientation': {'w': 1}}}
```
How are these used? These files look like the inputs to the task state machine, so a usage note in the readme seems appropriate.

nickswalker reviewed (changes requested):
First round of comments based on reading code only

nickswalker commented:
Posting screenshot for now, will update with comments on the state machine itself later
![Screenshot from 2020-06-05 15-57-27](https://user-images.githubusercontent.com/924295/83925783-a1f91f00-a74d-11ea-9edf-b3476cc68347.png)

nickswalker committed:
Fix path library path env-hooks

Turns out to be tricky to use file globs because these hooks have to run in dash
as a part of catkin build. Switched to grep

nickswalker committed:
Add task state machine loading test

Confirms that the state machine is loadable
Disabled for until DLR-RM/RAFCON#838 is resolved

nickswalker committed:
Fix state machine load test

Required a minor workaround to library behavior when states are missing

nickswalker committed:
Drop unused files

Drop dead references in demo launch

nickswalker committed:
Clean up navigation states

Rename to use "go_to" as a self-explanatory prefix
Add teleportation navigation simulation fallback

nickswalker committed:
Clean up initialization and surface scan input params

Set a simulation flag in initialization
Load map from knowledgebase during initialization
Tuck manipulation manager spinup behind option

nickswalker committed:
Add helper for running/testing individual states

nickswalker committed:
Update library state usage

nickswalker committed:
Fix run_task shebang

nickswalker committed:
Drop editor files

nickswalker committed:
Fix PEP8 violations

nickswalker committed:
Minor fixes to movement states

Add new move_to_go for tucking and resetting torso
Don't move any joints in initialization. Mostly to make fast re-initialization possible for debugging, but also because motion is slow and needs to be opt in
Implement basic gripper state based check_grasped

nickswalker committed:
Attempt to fix realsense_ros CI build issues

librealsense and the wrapper seem to be out of version sync

nickswalker committed:
Fix rosdep keys

nickswalker committed:
Fix rosdep keys

nickswalker committed:
Manually install rafcon in CI

nickswalker committed:
Rename basic manipulation states
